### PR TITLE
Bump TypeScript

### PR DESCRIPTION
**Pull request by ganymedes01**
TypeScript seems to have changed how it handles union types with repeating properties. While it still works for the case above in most cases (`a` will correctly be `toot | doot`), the previous way `safe` used to infer all keys from the union stopped working

```ts
type MyType = { a: 'toot'; b: number } | { a: 'doot'; c: number };

function getAB(): MyType {
  return {} as any;
}

const obj = getAB();
safe(obj).a.$;
```

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -7,12 +7,10 @@ interface SafeProxy<Initial, To> {
   $pmap: (f: (t: To) => Promise<To>) => Promise<Initial>;
 }
 
-type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void
-  ? I
-  : never;
+type KeysOfUnion<T> = T extends any ? keyof T : never;
```
Comment (ganymedes01) [inline index.ts:10]: By using conditional types TypeScript will go through all types that make up `T` and return their keys 😬

Comment (sakari) [reply index.ts:10]: ...right. might be nice to have some @ts-expect-error tests for things so that we do not end up accepting everything.

Comment (ganymedes01) [reply index.ts:10]: What do we not want to accept?

Comment (sakari) [reply index.ts:10]: like `const a: { a: number}: safe(a).b.$` should be an error and stuff like that. Also maybe check that we follow the field types when the type is an union. Or that the resulting type is what you would expect.

**Pull request merged**